src/js/share.js

```diff
@@ -0,0 +1,251 @@
+		return templateUrl;
+	}
+
+	/**
+	 * Renders the list of social networks in {@link config.links}
+	 *
+	 * @private
+	 */
+	function render() {
+		var ulElement = document.createElement('ul');
+		for (var i = 0; i < config.links.length; i++) {
+			var liElement = document.createElement('li');
+			liElement.classList.add('o-share__action', 'o-share__action--'+config.links[i]);
+			var aElement = document.createElement('a');
+			aElement.href = generateSocialUrl(config.links[i]);
+			aElement.appendChild(document.createElement('i'));
+			liElement.appendChild(aElement);
+			ulElement.appendChild(liElement);
+		}
+		oShare.rootEl.appendChild(ulElement);
+	}
+
+	/**
+	 * Initialises the Share class, rendering the o-share element if it's empty with {@link config} options,
+	 * or from corresponding data attributes and sets up dom-delegates.
+	 * Dispatches 'oShare.ready' at the end
+	 */
+	function init() {
+		if (!rootEl) {
+			rootEl = document.body;
+		} else if (!(rootEl instanceof HTMLElement)) {
+			rootEl = document.querySelector(rootEl);
+		}
+
+		var rootDelegate = new DomDelegate(rootEl);
+		rootDelegate.on('click', handleClick);
+		rootEl.setAttribute('data-o-share--js', '');
+
+		oShare.rootDomDelegate = rootDelegate;
+		oShare.rootEl = rootEl;
+
+		if (rootEl.children.length === 0) {
+			if (!config) {
+				config = {};
+				config.links = rootEl.hasAttribute('data-o-share-links') ?
+					rootEl.getAttribute('data-o-share-links').split(' ') : [];
+				config.url = rootEl.getAttribute('data-o-share-url') || '';
+				config.title = rootEl.getAttribute('data-o-share-title') || '';
+				config.titleExtra = rootEl.getAttribute('data-o-share-titleExtra') || '';
+				config.summary = rootEl.getAttribute('data-o-share-summary') || '';
+				config.relatedTwitterAccounts = rootEl.getAttribute('data-o-share-relatedTwitterAccounts') || '';
+			}
+			render();
+		}
+
+		dispatchCustomEvent('oShare.ready', {
+			share: oShare
+		});
+	}
+
+	init();
+}
+
+/**
+ * Destroys the Share instance, disables dom-delegates
+ */
+Share.prototype.destroy = function() {
+	this.rootDomDelegate.destroy();
+	// Should destroy remove its children? Maybe setting .innerHTML to '' is faster
+	for (var i = 0; i < this.rootEl.children; i++) {
+		this.rootEl.removeChild(this.rootEl.children[i]);
+	}
+
+	this.rootEl.removeAttribute('data-o-share--js');
+	this.rootEl = undefined;
+};
+
+/**
+ * Initialises all o-share components inside the element passed as the first parameter
+ *
+ * @param {(HTMLElement|string)} [el=document.body] - Element where to search for o-share components. You can pass an HTMLElement or a selector string
+ * @returns {Array} - An array of Share instances
+ */
+Share.init = function(el) {
+	var shareInstances = [];
+	var shareElements;
+
+	if (!el) {
+		el = document.body;
+	} else if (!(el instanceof HTMLElement)) {
+		el = document.querySelector(el);
+	}
+
+	if (el.querySelectorAll) {
```

Comment:
This doesn't guarantee that `querySelectorAll` will pass.

For example IE8 does have support for `querySelectorAll` but chokes on CSS3 selectors such as `:not()`.